DEPLOY.rst

```diff
@@ -0,0 +1,198 @@
+~~~~~~~~~~~
+
+Currently there's some manual setup required in the AWS console
+before we can start using the automation.
+
+Please work *carefully and thoroughly* through the
+`Fabulaws initial setup documentation <http://fabulaws.readthedocs.org/en/latest/initial-setup.html>`_, changing
+``myproject`` to ``opendebates``.  This will include creating security groups,
+creating a load balancer, and creating an autoscaling group.
+
+You can skip most of the *Project Configuration* section because that's
+already been done for this project, except for
+`SSH Keys <http://fabulaws.readthedocs.org/en/latest/initial-setup.html#ssh-keys>`_.
+
+Changes to project files
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In ``fabulaws-config.yml`` in this project, find the section that
+looks like this::
+
+    site_domains_map:
+      production:
+      - dualstack.myproject-production-1-12345.us-east-1.elb.amazonaws.com
+      staging:
+      - dualstack.myproject-staging-1-12345.us-east-1.elb.amazonaws.com
+      testing:
+      - dualstack.myproject-testing-1-12345.us-east-1.elb.amazonaws.com
+
+and change the domain under the environment you're setting up to the
+full hostname of the load balancer you just created.
+
+Also look for this section::
+
+      auto_scaling_groups:
+        opendebates:
+          production: opendebates-production-ag
+          staging: opendebates-staging-ag
+          testing: opendebates-testing-ag
+
+and change the appropriate value to the name of the autoscaling group you
+created.  (Or create it with that name to begin with.)
+
+Automation commands
+~~~~~~~~~~~~~~~~~~~
+
+Now you can run this command to set up everything else::
+
+    fab create_environment:opendebates,<environment>
+
+It'll take a while (30 minutes? an hour?) but you only need to do it
+once per environment.
+
+Updating code
+-------------
+
+This is what to do when the code changes and you want the servers in
+an environment to switch to the newer code.
+
+Read through this whole section before starting to update anything,
+please!
+
+Step 1: Create a new launch configuration. This is a saved EC2 instance image
+that the autoscaler uses to spin up new web servers::
+
+     fab create_launch_config_for_deployment:opendebates,<environment>
+
+On an m1.small instance, this'll take just over 20 minutes. It might be faster
+if your web servers are using a faster server.
+
+At the end of the output of that command, it'll print out a long string that
+is the name of the new launch configuration. Save that somewhere.
+
+Now, which order you do steps 2 and 3 in might depend on whether you
+have migrations and whether it's okay for the migration to run before
+the web servers have been updated, or for the web servers to be updated
+before the migrations are run, or whether they really really need to happen
+at the same time.
+
+If you can run migrations first, or don't have any, do step 2, then step 3.
+
+If you have to update webservers first, do step 3, then step 2.
```

Comment:
Since deploy_full and deploy_serial both run deploy_worker, this will never happen (and mustn't, esp. when using remote static media, since the worker *must* deploy the latest compressed media before the web servers are updated to use it).